src/onboarding/onboarding-restore-snapshot.ts

```diff
@@ -0,0 +1,160 @@
+  TemplateResult,
+} from "lit-element";
+import "../../hassio/src/components/hassio-ansi-to-html";
+import { showHassioSnapshotDialog } from "../../hassio/src/dialogs/snapshot/show-dialog-hassio-snapshot";
+import { showSnapshotUploadDialog } from "../../hassio/src/dialogs/snapshot/show-dialog-snapshot-upload";
+import { navigate } from "../common/navigate";
+import type { LocalizeFunc } from "../common/translations/localize";
+import "../components/ha-card";
+import { makeDialogManager } from "../dialogs/make-dialog-manager";
+import { ProvideHassLitMixin } from "../mixins/provide-hass-lit-mixin";
+import { haStyle } from "../resources/styles";
+
+declare global {
+  interface HASSDomEvents {
+    restoring: undefined;
+  }
+}
+
+@customElement("onboarding-restore-snapshot")
+class OnboardingRestoreSnapshot extends ProvideHassLitMixin(LitElement) {
+  @property() public localize!: LocalizeFunc;
+
+  @property() public language!: string;
+
+  @property({ type: Boolean }) private restoring = false;
+
+  @internalProperty() private _log?: string;
+
+  protected render(): TemplateResult {
+    return this.restoring
+      ? html`<ha-card>
+          <h2>
+            ${this.localize("ui.panel.page-onboarding.restore.in_progress")}
+          </h2>
+          ${this._log
+            ? html`<hassio-ansi-to-html
+                class="log"
+                .content=${this._filterLogs(this._log)}
+              ></hassio-ansi-to-html>`
+            : ""}
+        </ha-card>`
+      : html`
+          <button class="link" @click=${this._uploadSnapshot}>
+            ${this.localize("ui.panel.page-onboarding.restore.description")}
+          </button>
+        `;
+  }
+
+  private _filterLogs(logs: string): string {
+    // Filter out logs that is not relevant to show during the restore
+    return logs
+      .split("\n")
+      .filter(
+        (entry) =>
+          !entry.includes("/supervisor/logs") &&
+          !entry.includes("/supervisor/ping") &&
+          !entry.includes("DEBUG")
+      )
+      .slice(-16, -1) // We only show the last 15 lines, anything beyond that is not a part of the restore
+      .join("\n")
+      .replace(/\s[A-Z]+\s\(\w+\)\s\[[\w.]+\]/gi, "")
+      .replace(/\d{2}-\d{2}-\d{2}\s/gi, "");
+  }
+
+  private _uploadSnapshot(): void {
+    showSnapshotUploadDialog(this, {
+      showSnapshot: (slug: string) => this._showSnapshotDialog(slug),
+      onboarding: true,
+    });
+  }
+
+  protected firstUpdated(changedProps) {
+    super.firstUpdated(changedProps);
+    makeDialogManager(this, this.shadowRoot!);
+    setInterval(() => this._getLogs(), 1000);
+  }
+
+  private async _getLogs(): Promise<void> {
+    if (this.restoring) {
+      try {
+        const response = await fetch("/api/hassio/supervisor/logs", {
+          method: "GET",
+        });
+        const logs = await response.text();
+        this._log = this._filterLogs(logs);
+        if (this._log.match(/\d{2}:\d{2}:\d{2}\s.*Restore\s\w+\sdone/)) {
```

Comment:
Could we do this differently? Seems like a hack to check the logs for a text

Reply:
Not currently, but when we have <https://github.com/home-assistant/supervisor/issues/1946> we might